shared/src/business/useCases/docketEntry/addPaperFilingInteractor.ts

```diff
@@ -75,85 +98,126 @@ export const addPaperFilingInteractor = async (
   }
 
   const servedParties = aggregatePartiesForService(caseEntity);
+
   const docketRecordEditState =
     metadata.isFileAttached === false ? documentMetadata : {};
 
-  const docketEntryEntity = new DocketEntry(
-    {
-      ...baseMetadata,
-      ...metadata,
-      docketEntryId,
-      documentTitle: metadata.documentTitle,
-      documentType: metadata.documentType,
-      editState: JSON.stringify(docketRecordEditState),
-      filingDate: metadata.receivedAt,
-      isOnDocketRecord: true,
-      mailingDate: metadata.mailingDate,
-      relationship,
-      userId: user.userId,
-    },
-    { applicationContext, petitioners: caseEntity.petitioners },
-  );
-
-  const workItem = new WorkItem(
-    {
-      assigneeId: null,
-      assigneeName: null,
-      associatedJudge: caseToUpdate.associatedJudge,
-      caseStatus: caseToUpdate.status,
-      caseTitle: Case.getCaseTitle(caseEntity.caseCaption),
-      docketEntry: {
-        ...docketEntryEntity.toRawObject(),
-        createdAt: docketEntryEntity.createdAt,
+  let caseEntities = [];
+  //TODO sorry for ugly
+  for (let docketNo of consolidatedGroupDocketNumbers) {
+    const aCase = await applicationContext
+      .getPersistenceGateway()
+      .getCaseByDocketNumber({
+        applicationContext,
+        docketNumber: docketNo,
+      });
+
+    let aCaseEntity = new Case(aCase, { applicationContext });
+    caseEntities.push(aCaseEntity);
+  }
+
+  let filedByFromLeadCase;
+  //why raw case? who knows. couldnt think of a name
+  for (const rawCase of caseEntities) {
+    const docketEntryEntity = new DocketEntry(
+      {
+        ...baseMetadata,
+        ...metadata,
+        docketEntryId,
+        documentTitle: metadata.documentTitle,
+        documentType: metadata.documentType,
+        editState: JSON.stringify(docketRecordEditState),
+        filingDate: metadata.receivedAt,
+        isOnDocketRecord: true,
+        mailingDate: metadata.mailingDate,
+        relationship,
+        userId: user.userId,
       },
-      docketNumber: caseToUpdate.docketNumber,
-      docketNumberWithSuffix: caseToUpdate.docketNumberWithSuffix,
-      inProgress: isSavingForLater,
-      isRead: user.role !== ROLES.privatePractitioner,
+      { applicationContext, petitioners: rawCase.petitioners },
+    );
+
+    if (rawCase.docketNumber === rawCase.leadDocketNumber) {
```

Comment:
do you think we should just have a case entity method called `isLeadCase()` which does this logic?

Reply:
Good idea! You okay with us adding that as a DOD refactoring item, since we still have some stuff to do for this story?